lib/Froxlor/Api/Commands/Mysqls.php

```diff
@@ -49,13 +49,28 @@ public function add()
 
 			// parameters
 			$dbserver = $this->getParam('mysql_server', true, 0);
-			$databasedescription = $this->getParam('description', true, '');
+			$databasedescription = $originaldatabasedescription = $this->getParam('description', strtoupper(Settings::Get('customer.mysqlprefix')) == 'DBNAME' ? false : true, '');
 			$sendinfomail = $this->getBoolParam('sendinfomail', true, 0);
 
 			// validation
 			$password = \Froxlor\Validate\Validate::validate($password, 'password', '', '', array(), true);
 			$password = \Froxlor\System\Crypt::validatePassword($password, true);
-			$databasedescription = \Froxlor\Validate\Validate::validate(trim($databasedescription), 'description', '', '', array(), true);
+
+            // get needed customer info to reduce the mysql-usage-counter by one
+            $customer = $this->getCustomerData('mysqls');
+			$newdbname = '';
+			if(strtoupper(Settings::Get('customer.mysqlprefix')) == 'DBNAME') {
+			    //If DBNAME is used for prefix, we add customername + id + fieldescription to the validator.
+                //Also we pass in a regex, so we prevent invalid dbnames for mysql.
+                //read more @ https://dev.mysql.com/doc/refman/8.0/en/identifiers.html
+                $newdbname = ($this->isAdmin() ? $customer['loginname'] : $this->getUserDetail('loginname')) . '_' . trim($databasedescription);
+                $databasedescription = \Froxlor\Validate\Validate::validate($newdbname, 'description', '/^[0-9a-zA-Z$_]{1,64}$/', '', array(), true);
```

Comment:
shouldn't that be:
```php
$newdbname = \Froxlor\Validate\Validate::validate($newdbname, 'description', '/^[0-9a-zA-Z$_]{1,64}$/', '', array(), true);
```
Which automatically makes the variable `$originaldatabasedescription` superfluous

Reply:
Not sure anymore why I made this, I've removed it.